### PR TITLE
Added API for user details and fixed naming conventions

### PR DESCRIPTION
**Pull request by 201560543**

### CODE REVIEW

File: app/web/api/api_filters.py [hunk 1]
```diff
@@ -2,18 +2,20 @@
 from datetime import datetime
 
 
-def date_range_filter(obj, filter_key ,start_date=None, end_date=None):
+def date_range_filter(obj, filter_key, start_date=None, end_date=None):
     """
         Utility function to get the date ranger filter for the passed query object.
     """
+    new_start_date = datetime.strptime(start_date, DB_DATE_FORMAT).date() if start_date else None
```
Comment (PansulBhatt) [inline app/web/api/api_filters.py:9]: Thanks for clearing this out. @pdeguzman96 Is this how you envisioned it as well?

Comment (pdeguzman96) [reply app/web/api/api_filters.py:9]: Yup, looks good

Comment (pdeguzman96) [reply app/web/api/api_filters.py:9]: Yeah this looks great

File: app/web/api/api_utils.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def converter(obj):
     elif isinstance(obj, date):
         return obj.__str__()
     elif isinstance(obj, Decimal):
-        return obj.__str__()
+        return float(obj)
```
Comment (pdeguzman96) [inline app/web/api/api_utils.py:18]: Good catch

File: app/web/api/api_utils.py [hunk 3]
```diff
@@ -51,6 +51,35 @@ def inner(*args, **kwargs):
         return inner
     return handler
 
+def check_parameters(required_params:list=[]):
```
Comment (pdeguzman96) [inline app/web/api/api_utils.py:54]: This is nice!

File: app/web/api/purchase_routes.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+from flask import current_app, request, abort
+from datetime import datetime as dt
+from web.preprocessor.constants import DB_DATE_FORMAT
+from web.auth import requires_auth
+
+@purchase.route('/', methods=['GET'])
+@exception_handler(custom_msg="Issues in fetching purchases")
+@check_parameters(['account_num', 'categories', 'suppliers'])
+@requires_auth
+def get_all_purchases():
```
Comment (pdeguzman96) [inline app/web/api/purchase_routes.py:16]: Good work on this

**Pull request merged**